### ISSUE TITLE
[BUG] when joining the same table for different columns displays repeated results

### ISSUE THREAD

**Issue by Imeguras** [cweijan/vscode-mysql#468]
- OS: Arch Linux x86_64
- Database Type: mysql/ mariasql ( mysql  Ver 15.1 Distrib 10.6.5-MariaDB, for Linux (x86_64) using readline 5.1)

so basically im doing a private project and it has boogaloo business logic as i want to keep track of all my valuables (which have different constraints and rules) in the same database but doing this:
```sql
SELECT * FROM ledger
inner join fi_generic on ledger.idInstrument = fi_generic.id
inner join fi_currency on fi_generic.id=fi_currency.id
inner join fit_generic as r1 on fi_currency.acquired_pair=r1.id
inner join fit_generic as s1 on fi_currency.acquisition_pair=s1.id
where idUser='1';
```
Results in the following on the extension
[https://u.teknik.io/cq8Ff.png](png)
but on the mysql cli it correctly responds
```
MariaDB [makr]> SELECT * FROM ledger
    -> inner join fi_generic on ledger.idInstrument = fi_generic.id
    -> inner join fi_currency on fi_generic.id=fi_currency.id
    -> inner join fit_generic as r1 on fi_currency.acquired_pair=r1.id
    -> inner join fit_generic as s1 on fi_currency.acquisition_pair=s1.id
    -> where idUser='1';
+--------+--------------+----+------------+------------+----------+-------+----+---------------------+------------+----------+-------------------+------------------+--------------+-------------+---------------+----+---------------------+------------+-----------+---------------------+--------+--------------+----+---------------------+------------+------------+---------------------+--------+--------------+
| idUser | idInstrument | id | created_at | updated_at | fi_tag   | id_fi | id | created_at          | updated_at | quantity | acquisition_price | acquisition_pair | barter_price | barter_pair | acquired_pair | id | created_at          | updated_at | name      | description         | suffix | isFractional | id | created_at          | updated_at | name       | description         | suffix | isFractional |
+--------+--------------+----+------------+------------+----------+-------+----+---------------------+------------+----------+-------------------+------------------+--------------+-------------+---------------+----+---------------------+------------+-----------+---------------------+--------+--------------+----+---------------------+------------+------------+---------------------+--------+--------------+
|      1 |            1 |  1 | NULL       | NULL       | Currency |     1 |  1 | 2021-04-28 11:38:42 | NULL       |        5 |                 2 |                4 |         NULL |        NULL |             3 |  3 | 2021-04-23 06:52:43 | NULL       | US Dollar | Currencies are fiat | USD    |            1 |  4 | 2021-10-24 17:17:29 | NULL       | Euro       | Currencies are fiat | EUR    |            1 |
|      1 |            2 |  2 | NULL       | NULL       | Currency |     2 |  2 | 2021-03-01 02:36:37 | NULL       |       10 |                40 |                2 |         NULL |        NULL |             3 |  3 | 2021-04-23 06:52:43 | NULL       | US Dollar | Currencies are fiat | USD    |            1 |  2 | 2021-08-23 02:21:43 | NULL       | Withdrawal | Generated to IRL    | 0xWTH  |            1 |
|      1 |            3 |  3 | NULL       | NULL       | Currency |     3 |  3 | 2021-08-03 11:01:20 | NULL       |        5 |                 2 |                2 |         NULL |        NULL |             3 |  3 | 2021-04-23 06:52:43 | NULL       | US Dollar | Currencies are fiat | USD    |            1 |  2 | 2021-08-23 02:21:43 | NULL       | Withdrawal | Generated to IRL    | 0xWTH  |            1 |
|      1 |            4 |  4 | NULL       | NULL       | Currency |     4 |  4 | 2021-04-06 01:09:46 | NULL       |       10 |                40 |                4 |         NULL |        NULL |             3 |  3 | 2021-04-23 06:52:43 | NULL       | US Dollar | Currencies are fiat | USD    |            1 |  4 | 2021-10-24 17:17:29 | NULL       | Euro       | Currencies are fiat | EUR    |            1 |
|      1 |            6 |  6 | NULL       | NULL       | Currency |     6 |  6 | 2021-11-08 19:23:49 | NULL       |       10 |                40 |                4 |         NULL |        NULL |             3 |  3 | 2021-04-23 06:52:43 | NULL       | US Dollar | Currencies are fiat | USD    |            1 |  4 | 2021-10-24 17:17:29 | NULL       | Euro       | Currencies are fiat | EUR    |            1 |
|      1 |            8 |  8 | NULL       | NULL       | Currency |     8 |  8 | 2021-09-20 07:31:35 | NULL       |       10 |                40 |                4 |         NULL |        NULL |             3 |  3 | 2021-04-23 06:52:43 | NULL       | US Dollar | Currencies are fiat | USD    |            1 |  4 | 2021-10-24 17:17:29 | NULL       | Euro       | Currencies are fiat | EUR    |            1 |
+--------+--------------+----+------------+------------+----------+-------+----+---------------------+------------+----------+-------------------+------------------+--------------+-------------+---------------+----+---------------------+------------+-----------+---------------------+--------+--------------+----+---------------------+------------+------------+---------------------+--------+--------------+
6 rows in set (0.000 sec)

MariaDB [makr]> 

```
TL;DR my best guess is that the graphical interface is probably cutting corners and making a shortcut displaying the same result twice since after all the inner join references the same table on the same respective column, its just that i refer a different columns on fi_currency

**Comment by cweijan** [comment]
I have stated this on the readme.
![image](https://user-images.githubusercontent.com/27798227/153734809-decbef58-c5d7-4062-8467-2e267ba1b5d0.png)
